### PR TITLE
[13.0][IMP] delivery_package_number: Hide Number of packages in Immediate Transfer wizard if picking have not carrier.

### PR DESCRIPTION
**Pull request by victoralmau**
Hide Number of packages in Immediate Transfer wizard if picking have not carrier.

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT38549

### CODE REVIEW

File: delivery_package_number/wizard/stock_inmediate_transfer.py [hunk 1]
```diff
@@ -9,6 +9,14 @@ class StockImmediateTransfer(models.TransientModel):
...
         help="Set the number of packages for this picking(s)",
     )
+    ask_number_of_packages = fields.Boolean(compute="_compute_ask_number_of_packages")
+
+    @api.depends("pick_ids")
+    def _compute_ask_number_of_packages(self):
+        for item in self:
+            item.ask_number_of_packages = fields.first(
+                item.pick_ids
+            ).picking_type_id.ask_number_of_packages
```
Comment (chienandalu) [inline delivery_package_number/wizard/stock_inmediate_transfer.py]:
```suggestion
        if len(self.pick_ids) != 1:
            return False
        return bool(self.pick_ids.carrier_id)
```

**Review by chienandalu — COMMENTED** [review]
I'd say it would be enough to just check if there's a carrier assigned. Maybe there's no need to overbloat picking type config with this... for now :)

**Comment by victoralmau** [comment]
> I'd say it would be enough to just check if there's a carrier assigned. Maybe there's no need to overbloat picking type config with this... for now :)

Changes done according to this (more optimal) approach.

File: delivery_package_number/wizard/stock_inmediate_transfer.py [hunk 2]
```diff
@@ -9,6 +9,14 @@ class StockImmediateTransfer(models.TransientModel):
     number_of_packages = fields.Integer(
         help="Set the number of packages for this picking(s)",
     )
+    ask_number_of_packages = fields.Boolean(compute="_compute_ask_number_of_packages")
+
+    @api.depends("pick_ids")
```
Comment (chienandalu) [inline delivery_package_number/wizard/stock_inmediate_transfer.py]: Not sure if this is needed, as pick_ids won't change in the lifetime of the wizard...

File: delivery_package_number/static/description/index.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
-<meta name="generator" content="Docutils 0.15.1: http://docutils.sourceforge.net/" />
+<meta name="generator" content="Docutils: http://docutils.sourceforge.net/" />
```
Comment (pedrobaeza) [inline delivery_package_number/static/description/index.html]: Why this change?

File: delivery_package_number/wizard/stock_inmediate_transfer.py [hunk 4]
```diff
@@ -9,6 +9,13 @@ class StockImmediateTransfer(models.TransientModel):
     number_of_packages = fields.Integer(
         help="Set the number of packages for this picking(s)",
     )
+    ask_number_of_packages = fields.Boolean(compute="_compute_ask_number_of_packages")
+
+    def _compute_ask_number_of_packages(self):
+        for item in self:
+            if len(item.pick_ids) != 1:
+                item.ask_number_of_packages = False
+            item.ask_number_of_packages = bool(item.pick_ids.carrier_id)
```
Comment (pedrobaeza) [inline delivery_package_number/wizard/stock_inmediate_transfer.py]: If there's more than one picking, no error will happen as they are joined together in a recordset

```suggestion
            item.ask_number_of_packages = bool(item.pick_ids.carrier_id)
```

Comment (chienandalu) [reply delivery_package_number/wizard/stock_inmediate_transfer.py]: The thing is to avoid showing the package number in those cases as well.

Comment (pedrobaeza) [reply delivery_package_number/wizard/stock_inmediate_transfer.py]: Yes, and that's the case for my suggestion.

**Comment by victoralmau** [comment]
Changes done.

**Review by chienandalu — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
